src/main/java/com/cgdp/library/config/BookMapper.java

```diff
@@ -0,0 +1,21 @@
+package com.cgdp.library.config;
+
+import com.cgdp.library.dto.book.BookDTO;
+import com.cgdp.library.dto.book.CreateBookRequestDTO;
+import com.cgdp.library.entity.BookEntity;
+import org.mapstruct.Mapper;
+import org.mapstruct.Mapping;
+import org.mapstruct.factory.Mappers;
+import org.springframework.stereotype.Component;
+
+@Mapper
+@Component
+public interface BookMapper {
+
+    BookMapper INSTANCE = Mappers.getMapper(BookMapper.class);
+
+    @Mapping(source = "authorId", target = "authorEntity.id")
+    BookEntity mapToBookEntity(CreateBookRequestDTO requestDTO);
+
+    BookDTO mapToBookDTO(BookEntity bookEntity);
```

Comment:
It is not mapping the authorId
```suggestion
   @Mapping(source = "authorEntity.id", target = "authorId")
    BookDTO mapToBookDTO(BookEntity bookEntity);
```